// next-i18next/dist/commonjs/utils/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "addSubpath", {
  enumerable: true,
  get: function get() {
    return _addSubpath["default"];
  }
});
Object.defineProperty(exports, "consoleMessage", {
  enumerable: true,
  get: function get() {
    return _consoleMessage["default"];
  }
});
Object.defineProperty(exports, "isServer", {
  enumerable: true,
  get: function get() {
    return _isServer["default"];
  }
});
Object.defineProperty(exports, "lngFromReq", {
  enumerable: true,
  get: function get() {
    return _lngFromReq["default"];
  }
});
Object.defineProperty(exports, "lngPathCorrector", {
  enumerable: true,
  get: function get() {
    return _lngPathCorrector["default"];
  }
});
Object.defineProperty(exports, "lngsToLoad", {
  enumerable: true,
  get: function get() {
    return _lngsToLoad["default"];
  }
});
Object.defineProperty(exports, "redirectWithoutCache", {
  enumerable: true,
  get: function get() {
    return _redirectWithoutCache["default"];
  }
});
Object.defineProperty(exports, "removeSubpath", {
  enumerable: true,
  get: function get() {
    return _removeSubpath["default"];
  }
});
Object.defineProperty(exports, "subpathFromLng", {
  enumerable: true,
  get: function get() {
    return _subpathFromLng["default"];
  }
});
Object.defineProperty(exports, "subpathIsPresent", {
  enumerable: true,
  get: function get() {
    return _subpathIsPresent["default"];
  }
});
Object.defineProperty(exports, "subpathIsRequired", {
  enumerable: true,
  get: function get() {
    return _subpathIsRequired["default"];
  }
});

var _addSubpath = _interopRequireDefault(require("./add-subpath"));

var _consoleMessage = _interopRequireDefault(require("./console-message"));

var _isServer = _interopRequireDefault(require("./is-server"));

var _lngFromReq = _interopRequireDefault(require("./lng-from-req"));

var _lngPathCorrector = _interopRequireDefault(require("./lng-path-corrector"));

var _lngsToLoad = _interopRequireDefault(require("./lngs-to-load"));

var _redirectWithoutCache = _interopRequireDefault(require("./redirect-without-cache"));

var _removeSubpath = _interopRequireDefault(require("./remove-subpath"));

var _subpathFromLng = _interopRequireDefault(require("./subpath-from-lng"));

var _subpathIsPresent = _interopRequireDefault(require("./subpath-is-present"));

var _subpathIsRequired = _interopRequireDefault(require("./subpath-is-required"));
