// @pankod/pankod-cli/lib/modules/paths/nextjs.paths.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.nextjs = {
    actionDir: './src/Actions',
    compInterfaceDir: './src/Interfaces/Components',
    componentsDir: './src/Components',
    definitionsDir: './src/Definitions',
    filesDir: './files',
    interfaceDir: './src/Interfaces/index.ts',
    mockDir: '/dir',
    mockPagesDir: '/pages',
    mockSrcDir: '/src',
    pageInterfaceDir: './src/Interfaces/Pages',
    pagesDir: './pages',
    reducerDir: './src/Redux/Reducers',
    reduxInterfaceDir: './src/Interfaces/Redux/Store.d.ts',
    routesDir: './app/routes.js',
    storeDir: './src',
    templates: {
        addRouteTemplate: './lib/Templates/nextjs/Routes.mustache',
        compInterfaceIndex: './lib/Templates/nextjs/Interfaces/ComponentIndex.mustache',
        createInterfaceTempPath: './lib/Templates/nextjs/Interfaces/Component.mustache',
        pageInterfaceIndex: './lib/Templates/nextjs/Interfaces/PageIndex.mustache',
        storeImportInterface: './lib/Templates/nextjs/Interfaces/ReduxImport.mustache',
        storeImportInterfaceFormatted: './lib/Templates/nextjs/Interfaces/ReduxImportFormatted.mustache',
        storeInterface: './lib/Templates/nextjs/Interfaces/ReduxStore.mustache',
        classComponentTemplatePath: './lib/Templates/nextjs/Components/Class.mustache',
        componentIndexTemplatePath: './lib/Templates/nextjs/Components/index.mustache',
        stylePageTemplate: './lib/Templates/nextjs/Styles.mustache',
        funcComponentTemplate: './lib/Templates/nextjs/Components/Functional.mustache',
        reducerIndexTemplatePath: './lib/Templates/nextjs/Reducers/index.mustache',
        reducerTemplatePath: './lib/Templates/nextjs/Reducers/Reducer.mustache',
        actionConstTemplatePath: './lib/Templates/nextjs/Reducers/ActionConst.mustache',
        actionTemplatePath: './lib/Templates/nextjs/Reducers/Action.mustache',
        actionIndexTemplatePath: './lib/Templates/nextjs/Reducers/ActionIndex.mustache',
        reducerStoreTemplatePath: './lib/Templates/nextjs/Reducers/Store.mustache'
    }
};
