// @pankod/pankod-cli/lib/modules/paths/nextjs2.paths.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.nextjs2 = {
    actionDir: './src/Actions',
    compInterfaceDir: './src/Interfaces/Components',
    componentsDir: './src/Components',
    definitionsDir: './src/Definitions',
    actionConstsFileDir: './src/Definitions/ActionConsts/ActionConsts.ts',
    filesDir: './files',
    interfaceDir: './src/Interfaces/index.ts',
    mockDir: '/dir',
    mockPagesDir: '/pages',
    mockSrcDir: '/src',
    pageInterfaceDir: './src/Interfaces/Pages',
    pagesDir: './pages',
    reducerDir: './src/Redux/Reducers',
    reduxInterfaceDir: './src/Redux/IStore.d.ts',
    routesDir: './server/routes.ts',
    storeDir: './src',
    pageStyledDir: './src/Styled',
    templates: {
        addRouteTemplate: './lib/Templates/nextjs2/Routes.mustache',
        compInterfaceIndex: './lib/Templates/nextjs2/Interfaces/ComponentIndex.mustache',
        createInterfaceTempPath: './lib/Templates/nextjs2/Interfaces/Component.mustache',
        pageInterfaceIndex: './lib/Templates/nextjs2/Interfaces/PageIndex.mustache',
        storeImportInterface: './lib/Templates/nextjs2/Interfaces/ReduxImport.mustache',
        storeImportInterfaceFormatted: './lib/Templates/nextjs2/Interfaces/ReduxImportFormatted.mustache',
        storeInterface: './lib/Templates/nextjs2/Interfaces/ReduxStore.mustache',
        classComponentTemplatePath: './lib/Templates/nextjs2/Components/Class.mustache',
        componentIndexTemplatePath: './lib/Templates/nextjs2/Components/index.mustache',
        stylePageTemplate: './lib/Templates/nextjs2/Styles.mustache',
        funcComponentTemplate: './lib/Templates/nextjs2/Components/Functional.mustache',
        reducerIndexTemplatePath: './lib/Templates/nextjs2/Reducers/index.mustache',
        reducerTemplatePath: './lib/Templates/nextjs2/Reducers/Reducer.mustache',
        actionConstTemplatePath: './lib/Templates/nextjs2/Reducers/ActionConst.mustache',
        actionTemplatePath: './lib/Templates/nextjs2/Reducers/Action.mustache',
        actionIndexTemplatePath: './lib/Templates/nextjs2/Reducers/ActionIndex.mustache',
        reducerStoreTemplatePath: './lib/Templates/nextjs2/Reducers/Store.mustache',
        styledComponentsTemplatePath: './lib/Templates/nextjs2/StyledComponents.mustache',
        actionTestTemplatePath: './lib/Templates/nextjs2/Tests/ActionTest.mustache',
        reducerTestTemplatePath: './lib/Templates/nextjs2/Tests/ReducerTest.mustache',
        componentTestTemplatePath: './lib/Templates/nextjs2/Tests/ComponentTest.mustache'
    }
};
